src/Controller/ArticleController.php

```diff
@@ -83,53 +103,41 @@ public function addArticle(Request $request)
             $this->saveEntity($article);
 
             $this->addSuccessFlash(sprintf('Successfully saved new article "%s"', $article->getName()));
-            return $this->redirectToRoute('list_articles');
+            return $this->redirectToRoute('admin_list_articles');
         }
 
         return ['form' => $form->createView()] ;
     }
 
     /**
-     * @Route(
-     * "/admin/article/article_edit",
-     * name="edit_article",
-     * methods={"POST"},
-     * requirements={
-     *  "_format": "json"
-     *  }
-     * )
+     * @Route("/admin/article/edit/{id}", name="edit_article", requirements={"id"="\d+"})
+     * @Template("/admin/article/edit.html.twig")
      */
-    public function editArticleTitle(Request $request, ArticleRepository $articleRepository)
+    public function editJoy(Request $request, Article $article)
     {
-        $content = json_decode($request->getContent(), true);
-        if (isset($content['id']) && isset($content['name'])) {
-            $id = $content['id'];
-            $newName = $content['name'];
-            $article = $articleRepository->find($id);
-            if ($article) {
-                $article->setName($newName);
-                $this->saveEntity($article);
-
-                return new Response($article->getName());
-            } else {
-                throw $this->createNotFoundException('Article doesn\'t exist');
-            }
-        } else {
-            throw new BadRequestHttpException('Not valid request', null, 400);
+        $form = $this->createForm(ArticleFormType::class, $article);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $article = $form->getData();
+
+            $this->saveEntity($article);
+
+            $this->addSuccessFlash(sprintf('Successfully edited article "%s"', $article->getName()));
+            return $this->redirectToRoute('admin_list_articles');
         }
+
+        return ['form' => $form->createView()];
     }
 
     /**
-     * @Route("/admin/article/article_delete/{id}", name="delete_article")
+     * @Route("/admin/article/delete/{id}", name="delete_article", requirements={"id"="\d+"})
      */
     public function deleteArticle(Article $article)
     {
-        if (count($article->getEvents()) > 0) {
-            $this->addNoticeFlash('This article has associated events and can not be deleted');
-        } else {
-            $this->removeEntity($article);
-            $this->addSuccessFlash('Successfully deleted article');
-        }
-        return $this->redirectToRoute('list_articles');
+        $this->removeEntity($article);
+        $this->addSuccessFlash('Successfully deleted article');
```

Comment:
@daria-sieroshtan 
Language.
Also please add to your todo list error pages, including 404